src/main/java/org/seattlevoluntech/controllers/UserController.java

```diff
@@ -19,11 +20,23 @@
     @Autowired UsersRepository userRepository;
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-    @RequestMapping("/user")
+    @RequestMapping("/users")
     public List<User> users(final HttpServletRequest request) {
-      logger.info(request.getRemoteUser());
+        if (!request.getRemoteUser().isEmpty())
+            logger.info(request.getRemoteUser());
+        return Lists.newArrayList(userRepository.findAll());
+    }
 
-      return Lists.newArrayList(userRepository.findAll());
+    @RequestMapping(value = "/currentUser", method = RequestMethod.GET)
+    public User currentUserName(HttpServletRequest request) {
+        if (!request.getRemoteUser().isEmpty()){
+            List<User> users = Lists.newArrayList(userRepository.findAll());
```

Comment:
Instead of looping through users, I'd write a query that searches the DB by this field instead.

Reply:
Fixed it with a findByTokenId method.